app/assets/javascripts/controllers/cloud_network/cloud_network_form_controller.js

```diff
@@ -0,0 +1,61 @@
+ManageIQ.angular.app.controller('cloudNetworkFormController', ['$http', '$scope', 'cloudNetworkFormId', 'miqService', function($http, $scope, cloudNetworkFormId, miqService) {
+  $scope.cloudNetworkModel = { name: '' };
+  $scope.formId = cloudNetworkFormId;
+  $scope.afterGet = false;
+  $scope.modelCopy = angular.copy( $scope.cloudNetworkModel );
+  $scope.model = "cloudNetworkModel";
+
+  ManageIQ.angular.scope = $scope;
+
+  if (cloudNetworkFormId == 'new') {
+    $scope.cloudNetworkModel.name = "";
+    $scope.newRecord = true;
+    $scope.cloudNetworkModel.enabled= true;
+    $scope.cloudNetworkModel.external_facing = false;
+    $scope.cloudNetworkModel.shared = false;
+
+  } else {
+    miqService.sparkleOn();
+
+    $http.get('/cloud_network/cloud_network_form_fields/' + cloudNetworkFormId).success(function(data) {
+      $scope.afterGet = true;
+      $scope.cloudNetworkModel.name = data.name;
+      $scope.cloudNetworkModel.shared = data.shared;
+      $scope.cloudNetworkModel.enabled= data.enabled;
+      $scope.cloudNetworkModel.external_facing = data.external_facing;
+      $scope.cloudNetworkModel.provider_network_type = data.provider_network_type;
+      $scope.cloudNetworkModel.cloud_tenant_name = data.cloud_tenant_name;
+
+      $scope.modelCopy = angular.copy( $scope.cloudNetworkModel );
+      miqService.sparkleOff();
+    });
+  }
+
+  $scope.addClicked = function() {
+    miqService.sparkleOn();
+    var url = 'create/new' + '?button=add';
+    miqService.miqAjaxButton(url, true);
+  };
+
+  $scope.cancelClicked = function() {
+    miqService.sparkleOn();
+    if (cloudNetworkFormId == 'new') {
+      var url = '/cloud_network/create/new' + '?button=cancel';
+    } else {
+      var url = '/cloud_network/update/' + cloudNetworkFormId + '?button=cancel';
+    }
+    miqService.miqAjaxButton(url);
+  };
+
+  $scope.saveClicked = function() {
+    miqService.sparkleOn();
+    var url = '/cloud_network/update/' + cloudNetworkFormId + '?button=save';
+    miqService.miqAjaxButton(url, true);
```

Comment:
Please don't use `miqAjaxButton` with true (trying to remove in https://github.com/ManageIQ/manageiq/pull/11324) - pass `$scope.cloudNetworkModel` instead.